server/views/pages/visitors.njk

```diff
@@ -81,6 +81,53 @@
 
       <p><strong class="govuk-!-margin-right-4">Prisoner name:</strong> <span data-test="prisoner-name">{{ prisonerName }}</span></p>
 
+      <div class="prisoner-restrictions">
+
+        <h2 class="govuk-heading-m">Prisoner's active restrictions</h2>
+
+        {% if restrictions | length %}
+          {% set prisonerRestrictions = [] %}
+          {% for restriction in restrictions %}
+            {%- set prisonerRestrictions = (prisonerRestrictions.push([
+              {
+                html: '<span class="test-restrictions-type' + loop.index + ' moj-badge moj-badge--large visitor-restriction-badge visitor-restriction-badge--' + restriction.restrictionType + '">' + restriction.restrictionTypeDescription + '</span>'
+              },
+              {
+                text: restriction.comment,
+                classes: "test-restrictions-comment" + loop.index
+              },
+              {
+                text: restriction.startDate | formatDate,
+                classes: "test-restrictions-start-date" + loop.index
+              },
+              {
+                text: restriction.expiryDate | formatDate,
```

Comment:
I think you might need to add 'Not entered' if there's no expiry date?(Looking at the Prison API, neither start nor end date are required fields so perhaps both could actually be missing?) Could add a second test restriction in `visitors.test.ts` with no date(s) to cover this.